token-contracts/WorkerToken/src/main/java/finance/omm/score/WorkerTokenImpl.java

```diff
@@ -0,0 +1,150 @@
+        this.decimals.set(_decimals);
+        this.balances.set(Context.getCaller(), totalSupply);
+
+        Context.println(TAG + "| Started Worker Token:" + Context.getAddress());
+
+    }
+
+    public void onUpdate() {
+        Context.println(TAG + "| on update event:" + Context.getAddress());
+    }
+
+    @EventLog(indexed = 3)
+    public void Transfer(Address _from, Address _to, BigInteger _value,
+            byte[] _data) {
+    }
+
+    @External(readonly = true)
+    public String name() {
+        return "OMM " + TAG;
+    }
+
+    @External(readonly = true)
+    public String symbol() {
+        return WORKER_TOKEN_SYMBOL;
+    }
+
+    @External(readonly = true)
+    public BigInteger decimals() {
+        return this.decimals.get();
+    }
+
+    @External(readonly = true)
+    public BigInteger totalSupply() {
+        return this.totalSupply.get();
+    }
+
+    @External(readonly = true)
+    public BigInteger balanceOf(Address _owner) {
+        return this.balances.getOrDefault(_owner, ZERO);
+    }
+
+    @External
+    public void transfer(Address _to, BigInteger _value, @Optional byte[] _data) {
+        if (_data == null || _data.length == 0) {
+            _data = "None".getBytes();
+        }
+
+        this.wallets.add(_to);
+        this._transfer(Context.getCaller(), _to, _value, _data);
+    }
+
+    public void _transfer(Address _from, Address _to, BigInteger _value, byte[] _data) {
+
+        // Checks the sending value and balance.
+        if (_value.compareTo(ZERO) <= 0) {
+            Context.revert(TAG + ": Transferring value should be greater than zero");
+        }
+
+        BigInteger balanceFrom = this.balances.getOrDefault(_from, ZERO);
+        Context.println(TAG + "| transferring: " + _value + " - from sender balance: " + balanceFrom);
+
+        if (balanceFrom.compareTo(_value) < 0) {
+            Context.revert(TAG + " : Out of balance: " + _value);
+        }
+
+        this.balances.set(_from, balanceFrom.subtract(_value));
+        this.balances.set(_to, this.balances.getOrDefault(_to, ZERO).add(_value));
+
+        if (this.balanceOf(_from).equals(ZERO)) {
+            this.wallets.remove(_from);
+        }
+
+        if (_to.isContract()) {
+            // If the recipient is SCORE,
+            // then calls `tokenFallback` to hand over control.
+            Context.call(_to, "tokenFallback", _from, _value, _data);
+        }
+        // Emits an event log `Transfer`
+        this.Transfer(_from, _to, _value, _data);
+    }
+
+    @External(readonly = true)
+    public List<Address> getWallets() {
+        return arrayDbToList(this.wallets.getEntries());
+    }
+
+    private <T> List<T> arrayDbToList(ArrayDB<T> arraydb) {
+        @SuppressWarnings("unchecked")
+        T[] addressList = (T[]) new Object[arraydb.size()];
+
+        for (int i = 0; i < arraydb.size(); i++) {
+            addressList[i] = arraydb.get(i);
+        }
+        return List.of(addressList);
+    }
```

Comment:
I think this method can be moved to EnumerableSet itself with few following changes,
   ```
 public List<V> toList() {
        List<V> result = new ArrayList<>();
        int end = length() - 1;
        for (int i = 0; i < end; i++) {
            result.add(at(i));
        }
        return result;
    }
```

Reply:
end = length-1 
an then 
i < end ?

Reply:
hi @redlarva 
I have removed method above from WorkerTokenImpl class and moved it to EnumerableSet
cc: @cacomixtle